llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -561,6 +561,18 @@ createGCNMaxILPMachineScheduler(MachineSchedContext *C) {
   return DAG;
 }
 
+static ScheduleDAGInstrs *
+createGCNMaxMemoryClauseMachineScheduler(MachineSchedContext *C) {
+  const GCNSubtarget &ST = C->MF->getSubtarget<GCNSubtarget>();
+  ScheduleDAGMILive *DAG = new GCNScheduleDAGMILive(
+      C, std::make_unique<GCNMaxMemoryClauseSchedStrategy>(C));
+  DAG->addMutation(createLoadClusterDAGMutation(DAG->TII, DAG->TRI));
+  if (ST.shouldClusterStores())
+    DAG->addMutation(createStoreClusterDAGMutation(DAG->TII, DAG->TRI));
```

Comment:
Does this end up being redundant with the load and store clustering?

Reply:
Sorry I don't get the point here, why it is redundant? with what? (This is mostly inherited from the createGCNMaxOccupancyMachineScheduler())

Reply:
I mean if you cluster loads and stores, you're helping with clause formation

Reply:
Yes, it's true.